alacritty/src/display/mod.rs

```diff
@@ -491,6 +518,46 @@ impl Display {
         let vi_mode = terminal.mode().contains(TermMode::VI);
         let vi_mode_cursor = if vi_mode { Some(terminal.vi_mode_cursor) } else { None };
 
+        if self.collect_damage() {
+            let cursor = TermRenderableCursor { point: cursor_point, shape: cursor.shape() };
+            let terminal_damage = terminal.damage();
+
+            let requires_full_damage = self.visual_bell.intensity() != 0.
+                || self.hint_state.active()
+                || search_state.regex().is_some();
+            // Damage everything when search or hint picking is going.
+            if requires_full_damage {
+                terminal_damage.mark_fully_damaged();
+            }
+
+            // Add highlighted hints to damage information.
+            self.damage_hints(terminal_damage, display_offset);
+
+            let renderer_damage = RendererDamage::new(
+                terminal_damage,
+                self.size_info.into(),
+                cursor,
+                vi_mode_cursor,
+                selection_range,
+                display_offset,
+            );
+
+            // Collect renderer damage information and reset terminal damage. The reset is done
+            // automatically by an iterator.
+            for rect in renderer_damage {
+                self.damage_rects.push(rect)
+            }
+
+            // If we forced the entire screen to be damaged ensure that the next frame will be also
+            // fully damaged just to clean things up.
+            if requires_full_damage {
+                terminal_damage.mark_fully_damaged();
+            }
```

Comment:
Yes, but computing damage is cheap anyways and putting conditionals in functions which update damage will make it slower, since `min, max` are faster than `if false { max }`.